crates/core/src/articles/wallabag.rs

```diff
@@ -0,0 +1,574 @@
+use http::{HeaderValue, StatusCode};
+use regex::Regex;
+use serde::Deserialize;
+use std::{
+    fs::{self, File},
+    io::{Error, ErrorKind, Write},
+    sync::Arc,
+    time::{SystemTime, UNIX_EPOCH},
+};
+use ureq::Agent;
+use url::Url;
+
+use crate::{
+    articles::{Article, ArticleIndex, ARTICLES_DIR},
+    settings::ArticleAuth,
+    view::{Event, Hub},
+};
+
+struct ClientCredentials {
+    client_id: String,
+    client_secret: String,
+}
+
+// Create new API client by doing HTTP requests and reading the response HTML,
+// similar to how the Android app does it. If an API client with the given name
+// already exists, that API client is returned instead of creating a new one.
+//
+// This is a terrible idea, but there doesn't seem to be an alternative that
+// doesn't involve asking the user to manually create an API client and copying
+// over the client ID and secret. Since the Android app does something similar,
```

Comment:
Both Chrome extension and iOS app actually require creating an API client manually, so most of the users are probably used to that already.

Reply:
Oh you're right, it'd be a nightmare to type all the codes on the device.